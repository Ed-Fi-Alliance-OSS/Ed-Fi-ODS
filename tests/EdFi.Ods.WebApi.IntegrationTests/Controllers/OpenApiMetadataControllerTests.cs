// SPDX-License-Identifier: Apache-2.0
// Licensed to the Ed-Fi Alliance under one or more agreements.
// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
// See the LICENSE and NOTICES files in the project root for more information.

using System.Net;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;
using NUnit.Framework;
using Shouldly;

namespace EdFi.Ods.WebApi.IntegrationTests.Sandbox.Controllers
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter), typeof(NUnitReporter))]
    public class OpenApiMetadataControllerTests : HttpClientTestsBase
    {
        [Test]
        public async Task MetadataEndpointGetShouldBeValid()
        {
            var response = await HttpClient.GetAsync(TestConstants.BaseUrl + "metadata/");

            response.StatusCode.ShouldBe(HttpStatusCode.OK);

            var json = await response.Content.ReadAsStringAsync();

            json.ShouldNotBeNullOrWhiteSpace();
            Approvals.Verify(json, s => s.Replace(@"\r\n", @"\n"));
        }
    }
}