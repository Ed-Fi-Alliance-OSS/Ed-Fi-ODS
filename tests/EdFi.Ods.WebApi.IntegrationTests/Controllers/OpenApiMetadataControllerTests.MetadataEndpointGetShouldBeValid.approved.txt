[
  {
    "name": "Descriptors",
    "endpointUri": "http://localhost:5449/metadata/data/v3/descriptors/swagger.json",
    "prefix": ""
  },
  {
    "name": "Resources",
    "endpointUri": "http://localhost:5449/metadata/data/v3/resources/swagger.json",
    "prefix": ""
  }
]